encord/user_client.py

```diff
@@ -650,29 +664,119 @@ def create_project_from_cvat(
+                            CVAT_LONG_POLLING_MAX_REQUEST_TIME_SECONDS,
+                        ),
+                    ),
+                    result_type=CvatImportGetResultResponse,
+                )
+
+                if tmp_res.status == CvatImportGetResultLongPollingStatus.DONE:
+                    log.info(f"cvat import job completed with cvat_import_uuid={cvat_import_uuid}.")
+
+                polling_elapsed_seconds = ceil(time.perf_counter() - polling_start_timestamp)
+                polling_available_seconds = max(0, timeout_seconds - polling_elapsed_seconds)
+
+                if polling_available_seconds == 0 or tmp_res.status in [
+                    CvatImportGetResultLongPollingStatus.DONE,
+                    CvatImportGetResultLongPollingStatus.ERROR,
+                ]:
+                    res = tmp_res
+                    break
+
+                failed_requests_count = 0
+            except (requests.exceptions.RequestException, encord.exceptions.RequestException):
+                failed_requests_count += 1
+
+                if failed_requests_count >= CVAT_LONG_POLLING_RESPONSE_RETRY_N:
+                    raise
+
+                time.sleep(CVAT_LONG_POLLING_SLEEP_ON_FAILURE_SECONDS)
+
+        if res.status == CvatImportGetResultLongPollingStatus.DONE:
+            if res.project_uuid is None:
+                raise ValueError(f"{res.project_uuid=}, res.project_uuid should not be None with DONE status")
+
+            if res.issues is None:
+                raise ValueError(f"{res.issues=}, res.issues should not be None with DONE status")
+
             return CvatImporterSuccess(
-                project_hash=success["project_hash"],
-                dataset_hash=dataset_hash,
-                issues=Issues.from_dict(success["issues"]),
+                project_hash=str(res.project_uuid),
+                dataset_hash=str(list(self.get_project(res.project_uuid).list_datasets())[0]),
+                issues=Issues.from_dict(res.issues),
+            )
+        elif res.status == CvatImportGetResultLongPollingStatus.ERROR:
+            if res.issues is None:
+                raise ValueError(f"{res.issues=}, res.issues should not be None with DONE status")
+
+            return CvatImporterError(
+                issues=Issues.from_dict(res.issues),
             )
-        elif "error" in server_ret:
-            error = server_ret["error"]
-            return CvatImporterError(dataset_hash=dataset_hash, issues=Issues.from_dict(error["issues"]))
         else:
-            raise ValueError("The api server responded with an invalid payload.")
+            raise ValueError(f"{res.status=}, only DONE and ERROR status is expected after successful long polling")
+
+    def create_project_from_cvat(
+        self,
+        import_method: ImportMethod,
+        dataset_name: str,
+        review_mode: ReviewMode = ReviewMode.LABELLED,
+        max_workers: Optional[int] = None,
+        *,
```

Comment:
Curious as to what is the purpose of forcing the arguments after this to be keyword only?